libs/media-player/src/audio-annotate.cy.tsx

```diff
@@ -0,0 +1,234 @@
+
+                    cy.get('audio').then(([audioElement]) => {
+                        audioElement.currentTime = 1.0;
+                    });
+
+                    cy.getByDataAttribute('out-point-marker-button').click();
+
+                    cy.getByDataAttribute('time-range-selected').should('not.exist');
+
+                    cy.getByDataAttribute('audio-error-message').should('exist');
+
+                    // If I remove this cy.wait() the test suite fails at the first click
+                    cy.wait(2000);
+                });
+            });
+
+            describe(`after an in-point is selected and the clear range button is clicked`, () => {
+                it('should clear the selected in-point', () => {
+                    cy.wait(4000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+
+                    cy.wait(500);
+
+                    cy.getByDataAttribute('clear-selected-time-range-button').click();
+
+                    cy.getByDataAttribute('time-range-selected').should('not.exist');
+
+                    cy.getByDataAttribute('out-point-marker-button').should('be.disabled');
+                });
+            });
+
+            describe(`after a time range is selected and the clear range button is clicked`, () => {
+                it('should clear the range selection', () => {
+                    cy.wait(2000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+
+                    cy.wait(2500);
+
+                    cy.getByDataAttribute('out-point-marker-button').click();
+
+                    cy.getByDataAttribute('time-range-selected').should('exist');
+
+                    cy.wait(1000);
+
+                    cy.getByDataAttribute('clear-selected-time-range-button').click();
+
+                    cy.getByDataAttribute('time-range-selected').should('not.exist');
+                });
+            });
+
+            describe(`after a time range is selected with hotkeys and clear range is invoked with "c"`, () => {
+                it('should clear the range selection', () => {
+                    cy.wait(2000);
+
+                    cy.get('body').type('i');
+
+                    cy.wait(2500);
+
+                    cy.get('body').type('o');
+
+                    cy.getByDataAttribute('time-range-selected').should('exist');
+
+                    cy.wait(1000);
+
+                    cy.get('body').type('c');
+
+                    cy.getByDataAttribute('time-range-selected').should('not.exist');
+
+                    cy.getByDataAttribute('selection-bar-inpoint').should('not.be.visible');
+
+                    cy.getByDataAttribute('selection-bar-full').should('not.be.visible');
+                });
+            });
+        });
+    });
+
+    describe('given a valid audio url', () => {
+        describe(`when the mark in- and out-point buttons are clicked after the end of audio play`, () => {
```

Comment:
If the play  has finished do we want the out-point to be the end of the audio file? If not, how would a user mark a note that runs from X to the end of the audio?

Reply:
This is a case where the in point button is clicked after the end of play,  so the out point is the same as the in point, ie., equal to the duration of the media.  But in the scenario where the in point is already set before the end of play, the out point should be allowed to be equal to the duration.  This needs to be addressed in the `AudioAnnotator` component.